### ISSUE TITLE
Tests failing due to new pyright error

### ISSUE THREAD

**Issue by tsibley** [nextstrain/cli#143]
The pyright type check test failed last night with:

```
/home/runner/work/cli/cli/nextstrain/cli/types.py
  /home/runner/work/cli/cli/nextstrain/cli/types.py:27:20 - error: Use of "Protocol" requires Python 3.7 or newer
  /home/runner/work/cli/cli/nextstrain/cli/types.py:50:20 - error: Use of "Protocol" requires Python 3.7 or newer
2 errors, 0 warnings, 0 informations 
```

See also this example [in context of the full log](https://github.com/nextstrain/cli/runs/5047651539?check_suite_focus=true#step:7:107).

This is a new error from a new version of pyright, as our tests use the latest version and the previous CI runs were fine.

Confirmed locally that pyright version 1.1.211 passes but 1.1.217 does not.

**Comment by tsibley** [comment]
I think this is spurious?

We do ask pyright to check we're compat with 3.6

https://github.com/nextstrain/cli/blob/4ea39391901e92409912589fdd20e01ba8bb875d/pyrightconfig.json#L2

and `Protocol` was introduced in Python *3.8* as `typing.Protocol`, but we use the backported-to-3.6+ impl from `typing_extensions`.

Maybe pyright is saying that _it_ doesn't support checking `Protocol` anymore on 3.6 (maybe due to recent 3.6 EOL)?

**Comment by tsibley** [comment]
Fixed by #147.
